lib/Yield/PTML.cpp

```diff
@@ -535,19 +543,71 @@ static std::string taggedText(const PTMLBuilder &ThePTMLBuilder,
   return Result;
 }
 
+static std::string instructionPrefix(const PTMLBuilder &ThePTMLBuilder,
+                                     const yield::Instruction &Instruction) {
+  std::string Result;
+
+  if (options::InstructionAddressColumnWidth) {
+    std::string SerializedAddress = Instruction.Address().toString();
+    Result += ThePTMLBuilder.getTag(tags::Span, SerializedAddress)
+                .addAttribute(attributes::Token, tokenTypes::InstructionAddress)
+                .serialize()
+              + ' ';
+
+    if (SerializedAddress.size() < options::InstructionAddressColumnWidth) {
+      uint64_t Column = options::InstructionAddressColumnWidth;
+      Result += ThePTMLBuilder
+                  .getTag(tags::Span,
+                          std::string(Column - SerializedAddress.size(), ' '))
+                  .addAttribute(attributes::Token, ptml::tokens::Indentation)
+                  .serialize();
+    } else {
+      // TODO: handle column overflows more carefully.
+    }
+  }
+
+  if (options::RawByteColumnWidth) {
+    std::string SerializedBytes;
+    for (uint8_t Byte : Instruction.RawBytes()) {
+      std::string HexByte = Byte ? llvm::utohexstr(Byte, true, 2) : "00";
+      revng_assert(HexByte.size() == 2);
+      SerializedBytes += HexByte + ' ';
+    }
+
+    revng_assert(!SerializedBytes.empty());
+    Result += ThePTMLBuilder.getTag(tags::Span, SerializedBytes)
+                .addAttribute(attributes::Token, tokenTypes::RawBytes)
+                .serialize()
+              + ' ';
+
+    if (SerializedBytes.size() < options::RawByteColumnWidth) {
+      uint64_t Column = options::RawByteColumnWidth;
+      Result += ThePTMLBuilder
+                  .getTag(tags::Span,
+                          std::string(Column - SerializedBytes.size(), ' '))
+                  .addAttribute(attributes::Token, ptml::tokens::Indentation)
+                  .serialize();
+    } else {
+      // TODO: handle column overflows more carefully.
```

Comment:
So we don't emit anything in case there's too many bytes?

Reply:
If there are too many bytes, they "overflow", as in, instruction itself gets further to the right, breaking tabulation.

Reply:
But I'm fine with either option, we can emit nothing OR `byte byte byte...` or anything else. I don't have strong preferences.

Reply:
Ah this part emits spaces *after* the bytes. OK, maybe add a comment. I've to say that PTMLBuilder code is verbose and kinda hard to read.

In general I'd do what `objdump` does, but leaving this for later at this stage it's OK.

Reply:
Yeah, imitating `objdump` is non-trivial, since we'd need a way to introduce a "dummy" instructions that only have bytes, or fake it some other way. And then we need graph side to cooperate with that. And graph-side code is in a disarray already.

👌 on a comment.

Reply:
Since this got reworked, no reason to leave the comment anymore.